File: app/src/main/java/com/datarockets/mnchkn/activities/dashboard/DashboardInteractor.java
package com.datarockets.mnchkn.activities.dashboard;

import com.datarockets.mnchkn.models.Player;

import java.util.ArrayList;

public interface DashboardInteractor {

    interface OnLoadPlayerListener {
        void onFinished(ArrayList<Player> players);
        void onPlayerUpdated(Player player, int position);
    }

    interface OnScreenStatusListener {
        void onKeepScreenOn();
        void onKeepScreenOff();
    }

    void isScreenShouldBeOn(OnScreenStatusListener listener);
    void loadPlayersList(OnLoadPlayerListener listener);
    void updatePlayer(Player player, int position, OnLoadPlayerListener listener);
    void insertStep(Player player);
    void clearPlayersStats();
    void setGameFinished();
}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/dashboard/DashboardInteractor.kt
package com.datarockets.mnchkn.activities.dashboard

import com.datarockets.mnchkn.models.Player

import java.util.ArrayList

interface DashboardInteractor {

    interface OnLoadPlayerListener {
        fun onFinished(players: ArrayList<Player>)
        fun onPlayerUpdated(player: Player, position: Int)
    }

    interface OnScreenStatusListener {
        fun onKeepScreenOn()
        fun onKeepScreenOff()
    }

    fun isScreenShouldBeOn(listener: OnScreenStatusListener)
    fun loadPlayersList(listener: OnLoadPlayerListener)
    fun updatePlayer(player: Player, position: Int, listener: OnLoadPlayerListener)
    fun insertStep(player: Player)
    fun clearPlayersStats()
    fun setGameFinished()
}
